src/gt4py/next/program_processors/runners/dace_fieldview/transformations/auto_opt.py

```diff
@@ -0,0 +1,341 @@
+    """
+    return dace_aoptimize.auto_optimize(
+        sdfg,
+        device=device,
+        use_gpu_storage=use_gpu_storage,
+        **kwargs,
+    )
+
+
+def gt_simplify(
+    sdfg: dace.SDFG,
+    validate: bool = True,
+    validate_all: bool = False,
+    skip: Optional[set[str]] = None,
+) -> Any:
+    """Performs simplifications on the SDFG in place.
+
+    Instead of calling `sdfg.simplify()` directly, you should use this function,
+    as it is specially tuned for GridTool based SDFGs.
+
+    Args:
+        sdfg: The SDFG to optimize.
+        validate: Perform validation after the pass has run.
+        validate_all: Perform extensive validation.
+        skip: List of simplify passes that should not be applied.
+
+    Note:
+        The reason for this function is that we can influence how simplify works.
+        Since some parts in simplify might break things in the SDFG.
+        However, currently nothing is customized yet, and the function just calls
+        the simplification pass directly.
+    """
+    from dace.transformation.passes.simplify import SimplifyPass
+
+    return SimplifyPass(
+        validate=validate,
+        validate_all=validate_all,
+        verbose=False,
+        skip=skip,
+    ).apply_pass(sdfg, {})
+
+
+def gt_set_iteration_order(
+    sdfg: dace.SDFG,
+    leading_dim: gtx_common.Dimension,
+    validate: bool = True,
+    validate_all: bool = False,
+) -> Any:
+    """Set the iteration order of the Maps correctly.
+
+    Modifies the order of the Map parameters such that `leading_dim`
+    is the fastest varying one, the order of the other dimensions in
+    a Map is unspecific. `leading_dim` should be the dimensions were
+    the stride is one.
+
+    Args:
+        sdfg: The SDFG to process.
+        leading_dim: The leading dimensions.
+        validate: Perform validation during the steps.
+        validate_all: Perform extensive validation.
+    """
+    return sdfg.apply_transformations_once_everywhere(
+        gtx_transformations.MapIterationOrder(
+            leading_dim=leading_dim,
+        )
+    )
+
+
+def gt_auto_optimize(
+    sdfg: dace.SDFG,
+    gpu: bool,
+    leading_dim: Optional[gtx_common.Dimension] = None,
+    aggressive_fusion: bool = True,
+    make_persistent: bool = True,
+    gpu_block_size: Optional[Sequence[int | str] | str] = None,
+    block_dim: Optional[gtx_common.Dimension] = None,
+    blocking_size: int = 10,
+    reuse_transients: bool = False,
+    validate: bool = True,
+    validate_all: bool = False,
+    **kwargs: Any,
+) -> dace.SDFG:
+    """Performs GT4Py specific optimizations on the SDFG in place.
+
+    The auto optimization works in different phases, that focuses each on
+    different aspects of the SDFG. The initial SDFG is assumed to have a
+    very large number of rather simple Maps.
+
+    1. Some general simplification transformations, beyond classical simplify,
```

Comment:
There is actually no reason to not do it, however, if you look at the individual phases it does not _yet_ make sense to turn them into separate functions, with the exception of phase 2.

However, I agree that if the function become more complex we should separate the phases where it make sense.
And I will do it for the 2. phase.

Reply:
You said that I should not mark threads as resolved if I do not fully accept them.
It probably clicked on the wrong button or so.

But can _I_ mark this threads (or other threads such as this one) as resolved?

Reply:
In general I think you should only mark threads as resolved if you fully accept the suggestion from the reviewer and don't have further comments, so there is no need for the reviewer to look at it again. In the case you reply something, even if you agree with the suggestion, you should not mark the conversation as resolved because it will be collapsed and the reviewer won't see your comment.

Reply:
Does this mean that, If you made a comment and I replied something, then after the next round, the thread is still open, but you have not written anything to it, I can close it.